File: carcassonne-master/src/right_side_connections.cc
#include "right_side_connections.h"

namespace carcassonne
{

RightSideConnections::RightSideConnections()
{
}

RightSideConnections::RightSideConnections(const RightSideConnections& src)
	: SideConnections(src)
{
	CopyFrom(src);
}

RightSideConnections& RightSideConnections::
	operator=(const RightSideConnections& rhs)
{
	if(this == &rhs) {
		return (*this);
	}
	
	CopyFrom(rhs);
	
	return (*this);
}

void RightSideConnections::
	CopyFrom(const RightSideConnections& src)
{										  
}

SideConnections* RightSideConnections::
	Clone() const
{
	return new RightSideConnections(*this);
}

std::vector<std::string> RightSideConnections::ToStringVector() const
{
  std::vector<std::string> connections;

  connections.push_back("Right(itself)");

  if(clockwise_) {
    connections.push_back("Bottom");
  }

  if(across_) {
    connections.push_back("Left");
  }

  if(counterclockwise_) {
    connections.push_back("Top");
  }

  return connections;
  
}

RightSideConnections::~RightSideConnections()
{
}

}

File: js/RightSideConnections.js
/**
 * Created by marcogobbi on 10/10/2017.
 */
const SideConnections=require( "./SideConnections");
module.exports= class RightSideConnections extends SideConnections {
    constructor(src) {
        super(src);

    }

    Clone() {
        return new RightSideConnections(this);
    }

    ToStringVector() {
        var connections = [];

        connections.push("Right(itself)");

        if (this.clockwise_) {
            connections.push("Bottom");
        }

        if (this.across_) {
            connections.push("Left");
        }

        if (this.counterclockwise_) {
            connections.push("Top");
        }

        return connections;

    }

}